server/src/main/java/org/elasticsearch/tasks/Task.java

```diff
@@ -63,6 +64,10 @@ public class Task {
         X_ELASTIC_PRODUCT_ORIGIN_HTTP_HEADER
     );
 
+    public static Optional<String> extractTraceId(String traceparent) {
```

Comment:
`HttpTracer` is in a separate package, it currently doesn't access `RestController` at all. `Task` seemed like a sensible halfway house, as it has the header constants used by both.

Reply:
`RestController` is public though so you can just use it, I'd also rather see it there as well, `Task` really is the wrong place for this.

Reply:
Gotcha, moved.

Reply:
at the same time I would be careful putting utilities related to trace id in RestController. Maybe Task is not the best placce, but I would prefer to avoid to add utilities to RestController. It really does too much at the moment..

Reply:
perhaps a `RestUtil` would be better? Or a new util (:( ) with the header related methods?